compiler-rt/lib/scudo/standalone/flags.cpp

```diff
@@ -68,6 +69,13 @@ void initFlags() {
   Parser.parseString(getCompileDefinitionScudoDefaultOptions());
   Parser.parseString(getScudoDefaultOptions());
   Parser.parseString(getEnv("SCUDO_OPTIONS"));
+  if (const char *V = getEnv("SCUDO_ALLOCATION_RING_BUFFER_SIZE")) {
+    char *End = nullptr;
+    int I = static_cast<int>(strtol(V, &End, 10));
```

Comment:
Would it be better to leave this a long, and then do the cast down below?

Also, if you plan to allow this to be set, I think you need extra verification. For example, the number has to >= 0, and because the value is treated as a u32, you should make sure the number doesn't overflow that value. It might be better to have a maximum value for this.

Reply:
Added verification that this fits into an int here. Adding handling for < 0 in https://github.com/llvm/llvm-project/pull/71791. Will send another CL to limit size.